src/descriptor/mod.rs

```diff
@@ -661,9 +660,9 @@ where
 }
 
 impl Descriptor<DescriptorPublicKey> {
-    /// Derives all wildcard keys in the descriptor using the supplied `child_number`
-    pub fn derive(&self, child_number: bip32::ChildNumber) -> Descriptor<DescriptorPublicKey> {
-        self.translate_pk2_infallible(|pk| pk.clone().derive(child_number))
+    /// Derives all wildcard keys in the descriptor using the supplied index
+    pub fn derive(&self, index: u32) -> Descriptor<DescriptorPublicKey> {
```

Comment:
We should add a doccoment about index panic since this will function will be used a lot downstream.